docs/Python-API-Documentation.md

```diff
@@ -0,0 +1,1695 @@
+## ActionSpec
+
+```python
+class ActionSpec(NamedTuple)
+```
+
+A NamedTuple containing utility functions and information about the action spaces
+for a group of Agents under the same behavior.
+- num_continuous_actions is an int corresponding to the number of floats which
+constitute the action.
+- discrete_branch_sizes is a Tuple of int where each int corresponds to
+the number of discrete actions available to the agent on an independent action branch.
+
+<a name="mlagents_envs.base_env.ActionSpec.is_discrete"></a>
+#### is\_discrete
+
+```python
+ | is_discrete() -> bool
+```
+
+Returns true if this Behavior uses discrete actions
+
+<a name="mlagents_envs.base_env.ActionSpec.is_continuous"></a>
+#### is\_continuous
+
+```python
+ | is_continuous() -> bool
+```
+
+Returns true if this Behavior uses continuous actions
+
+<a name="mlagents_envs.base_env.ActionSpec.discrete_size"></a>
+#### discrete\_size
+
+```python
+ | @property
+ | discrete_size() -> int
+```
+
+Returns a an int corresponding to the number of discrete branches.
+
+<a name="mlagents_envs.base_env.ActionSpec.empty_action"></a>
+#### empty\_action
+
+```python
+ | empty_action(n_agents: int) -> ActionTuple
+```
+
+Generates ActionTuple corresponding to an empty action (all zeros)
+for a number of agents.
+
+**Arguments**:
+
+- `n_agents`: The number of agents that will have actions generated
+
+<a name="mlagents_envs.base_env.ActionSpec.random_action"></a>
+#### random\_action
+
+```python
+ | random_action(n_agents: int) -> ActionTuple
+```
+
+Generates ActionTuple corresponding to a random action (either discrete
+or continuous) for a number of agents.
+
+**Arguments**:
+
+- `n_agents`: The number of agents that will have actions generated
+
+<a name="mlagents_envs.base_env.ActionSpec.create_continuous"></a>
+#### create\_continuous
+
+```python
+ | @staticmethod
+ | create_continuous(continuous_size: int) -> "ActionSpec"
+```
+
+Creates an ActionSpec that is homogenously continuous
+
+<a name="mlagents_envs.base_env.ActionSpec.create_discrete"></a>
+#### create\_discrete
+
+```python
+ | @staticmethod
+ | create_discrete(discrete_branches: Tuple[int]) -> "ActionSpec"
+```
+
+Creates an ActionSpec that is homogenously discrete
+
+<a name="mlagents_envs.base_env.DimensionProperty"></a>
+## DimensionProperty
+
+```python
+class DimensionProperty(IntFlag)
+```
+
+No properties specified.
+
+<a name="mlagents_envs.base_env.DimensionProperty.UNSPECIFIED"></a>
+#### UNSPECIFIED
+
+No Property of the observation in that dimension. Observation can be processed with
+Fully connected networks.
+
+<a name="mlagents_envs.base_env.DimensionProperty.NONE"></a>
+#### NONE
+
+Means it is suitable to do a convolution in this dimension.
+
+<a name="mlagents_envs.base_env.DimensionProperty.TRANSLATIONAL_EQUIVARIANCE"></a>
+#### TRANSLATIONAL\_EQUIVARIANCE
+
+Means that there can be a variable number of observations in this dimension.
+The observations are unordered.
```

Comment:
This is straight from the docstrings. If that's the case, then the docstrings should be updated/fixed/formatted accordingly. I can't comment on the correctness of the docstrings, since I'm only just beginning to become familiar with the package.